### ISSUE TITLE
Meson failing due to libdl check on OpenBSD

### ISSUE THREAD

**Issue by brad0** [mpv-player/mpv#10901]
Trying to convert from WAF to Meson. This is the first issue I have run into..

```
===>  Configuring for mpv-0.35.0p1
The Meson build system
Version: 0.64.0
Source dir: /home/ports/pobj/mpv-0.35.0/mpv-0.35.0
Build dir: /home/ports/pobj/mpv-0.35.0/build-amd64
Build type: native build
Project name: mpv
Project version: 0.35.0
C compiler for the host machine: cc (clang 13.0.0 "OpenBSD clang version 13.0.0")
C linker for the host machine: cc ld.lld 13.0.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/usr/local/bin/python3)
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Run-time dependency libavcodec found: YES 58.134.100
Run-time dependency libavfilter found: YES 7.110.100
Run-time dependency libavformat found: YES 58.76.100
Run-time dependency libavutil found: YES 56.70.100
Run-time dependency libswresample found: YES 3.9.100
Run-time dependency libswscale found: YES 5.9.100
Run-time dependency libass found: YES 0.16.0
Run-time dependency threads found: YES
Compiler for C supports arguments -Werror=implicit-function-declaration: YES
Compiler for C supports arguments -Wno-error=deprecated-declarations: YES
Compiler for C supports arguments -Wno-error=unused-function: YES
Compiler for C supports arguments -Wempty-body: YES
Compiler for C supports arguments -Wdisabled-optimization: YES
Compiler for C supports arguments -Wstrict-prototypes: YES
Compiler for C supports arguments -Wno-format-zero-length: YES
Compiler for C supports arguments -Wno-redundant-decls: YES
Compiler for C supports arguments -Wvla: YES
Compiler for C supports arguments -Wno-format-truncation: NO
Compiler for C supports arguments -Wimplicit-fallthrough: YES
Compiler for C supports arguments -fno-math-errno: YES
Compiler for C supports arguments -Wformat -Werror=format-security: YES
Compiler for C supports link arguments -Wl,-z,noexecstack: YES
Compiler for C supports link arguments -Wl,--nxcompat,--no-seh,--dynamicbase: NO
Library dl found: NO

../../pobj/mpv-0.35.0/mpv-0.35.0/meson.build:335:0: ERROR: Feature cplugins cannot be enabled: cplugins not supported by the os or compiler!
```

It appears as if this is failing due to a check for libdl. OpenBSD does not have libdl.

**Comment by sfan5** [comment]
relevant infos and discussion of a solution: https://github.com/haasn/libplacebo/issues/147

**Comment by eli-schwartz** [comment]
This should have already been fixed by #9793 -- in fact, it was mpv that contributed the long-term fix to Meson that is mentioned there.

However it was broken by 1835dfc05c8298262dbf8e08a31d61bbfecdb30b

```diff
 libdl_dep = cc.find_library('libdl', required: false)
-libdl = cc.has_function('dlopen', dependencies: libdl_dep, prefix: '#include <dlfcn.h>')
-if libdl
+features += {'libdl': libdl_dep.found() and cc.has_function('dlopen', dependencies: libdl_dep, prefix: '#include <dlfcn.h>')}
+if features['libdl']
     dependencies += libdl_dep
 endif
-features += {'libdl': libdl}
```

The actual check for libdl works, but the feature is activated by the library check instead of just-the-symbol check.
